integreat_cms/deepl_api/utils.py

```diff
@@ -0,0 +1,87 @@
+import deepl
+import os
+
+from django.contrib import messages
+
+
+class DeepLApi:
+    """
+    DeepL API to auto translate selected posts.
+    """
+
+    def check_availability(self, request, language_slug):
+        """
+        this function checks, if the selected language is supported by DeepL
+
+        :param request: request that was sent to the server
+        :type request: ~django.http.HttpRequest
+
+        :param language_slug: current language slug
+        :type language_slug: str
+
+        :return: true or false
+        :rtype: boolean
+        """
+        deepl_langs = [
+            "bg",
+            "cs",
+            "da",
+            "de",
+            "el",
+            "en",
+            "es",
+            "et",
+            "fi",
+            "fr",
+            "hu",
+            "it",
+            "ja",
+            "lt",
+            "lv",
+            "nl",
+            "pl",
+            "pt",
+            "ro",
+            "ru",
+            "sk",
+            "sl",
+            "sv",
+            "zh",
+        ]
+        if language_slug in deepl_langs:
+            return language_slug in deepl_langs
+        else:
+            messages.warning(
+                request,
+                _("This language isn't supported by DeepL. Please try another one"),
+            )
+
+    def get_default_language(self, region):
+        """
+        # zu implementieren am 21.1.
+        This function gets the default language of the cms
+        """
+        default_language = region.default_language.slug
+        print(default_language)
+
+    def get_post(self, poi_ids):
+        """
+        # zu implementieren am 21.1.
+        # nachdem alle Infos zum Beitrag übergeben werden,
+        # wird nur der tatsächliche Inhalt des Posts geholt
+        """
+        print(poi_ids)
+
+    def get_translation(self, translator, post, target_language):
+        """
+        # zu implementieren am 24.1.
+        This function calls the translate-function of DeepL and
+        """
+        translator = deepl.Translator(os.getenv("de7e6874-aebd-0823-a278-c281a3760c64"))
```

Comment:
```suggestion
        translator = deepl.Translator(settings.DEEPL_AUTH_KEY)
```
When handling sensible data, it's usually a good idea to read the corresponding usage instructions:
> Create a Translator object providing your DeepL API authentication key.
> To avoid writing your key in source code, you can set it in an environment
> variable DEEPL_AUTH_KEY, then read the variable in your Python code:
> `translator = deepl.Translator(os.getenv("DEEPL_AUTH_KEY"))`

The reason not to write your secrets into source code is pretty simple: You start testing a few things and after that you forget about it and commit your source code to git - where the secret is now published... No worries, I already invalidated the token, so don't beat yourself up about it, but the correct workflow would be:
1. Add a new setting for this in `integreat_cms/core/settings.py`, e.g.: `DEEPL_AUTH_KEY = os.environ.get("INTEGREAT_CMS_DEEPL_AUTH_KEY")`
1. Read the secret into a environment variable, e.g. via the configuration file `/etc/integreat.ini` or via the command line: `read -rs INTEGREAT_CMS_DEEPL_AUTH_KEY && export INTEGREAT_CMS_DEEPL_AUTH_KEY` and then start the dev server which can now read this environment variable which is separated from the source code in the repository
2. In your DeepL api client, use `from django.conf import settings` and then you can use `deepl.Translator(settings.DEEPL_AUTH_KEY)` to initialize the translator object.